### PR TITLE
SKW-4168: backends cli

### PR DESCRIPTION
**Pull request by konvulse**
Functionally tested by adding backends and processing transactions in several environments.

### CODE REVIEW

**Comment by konvulse** [comment]
Removed Makefile change to avoid conflicts

File: conduce/api.py [hunk 1]
```diff
@@ -847,7 +849,31 @@ def list_dataset_backends(dataset_id, **kwargs):
     **kwargs : key-value
         See :py:func:`make_get_request` for more kwargs.
     """
-    return make_get_request('/api/v2/data/{}/backends'.format(dataset_id), **kwargs)
+    return json.loads(make_get_request('/api/v2/data/{}/backends'.format(dataset_id), **kwargs).content)
+
+
+def set_default_backend(dataset_id, backend_id, **kwargs):
```
Comment (konvulse) [inline conduce/api.py]: This new api method is in its own PR now.

File: conduce/api.py [hunk 2]
```diff
@@ -748,8 +748,10 @@ def get_transactions(dataset_id, **kwargs):
     # HACK: get around bool parameter bug
     if not kwargs.get('count'):
         parameters.pop('count')
+    if kwargs.get('max'):
+        parameters.pop('min')
```
Comment (jrmConduce) [inline conduce/api.py]: Why?

It is perfectly legal and reasonable to request a range of transactions, which is why a user would set both max and min.

File: conduce/api.py [hunk 3]
```diff
@@ -981,7 +1007,7 @@ def add_tile_store(dataset_id, auto_process, **kwargs):
     return _create_dataset_backend(dataset_id, DatasetBackends.tile, auto_process, None, **kwargs)
 
 
-def add_capped_tile_store(dataset_id, auto_process, min_temporal_zoom_level, min_spatial_zoom_level, **kwargs):
+def add_capped_tile_store(dataset_id, auto_process, min_spatial_zoom_level, min_temporal_zoom_level, **kwargs):
```
Comment (jrmConduce) [inline conduce/api.py]: No default values?

File: conduce/cli.py [hunk 4]
```diff
@@ -316,6 +316,169 @@ def get_dataset_transactions(args):
...
+
+    for backend_id in backend_ids:
+        location = api.remove_dataset_backend(dataset_id, backend_id, **vars(args)).headers['location']
+        print('Removing {}: check status at {}'.format(backend_id, location))
+
+
+def list_dataset_backends(args):
+    dataset_id = args.dataset_id
+    del vars(args)['dataset_id']
+    backends = args.backend_ids
```
Comment (jrmConduce) [inline conduce/cli.py]: It seems weird for a method that is supposed to list backends to optionally take a list of backends.

I get what you're going for, but it feels like that should be a separate method.

File: conduce/api.py (hunk 3 above)
Comment (jrmConduce) [reply conduce/api.py]: never mind this, I see the default is set below

File: conduce/api.py (hunk 2 above)
Comment (konvulse) [reply conduce/api.py]: Oops, that was not supposed to be included.  #160

File: conduce/cli.py (hunk 4 above)
Comment (konvulse) [reply conduce/cli.py]: yeah, and i suppose it only works with the `-v` option

Comment (konvulse) [reply conduce/cli.py]: I'll add a story to address this

Comment (konvulse) [reply conduce/cli.py]: SKW-4227

**Pull request merged**